tools/target-server-validator/README.md

```diff
@@ -0,0 +1,90 @@
+# Apigee Target Server Validator
+
+The objective of this tool to validate targets in Target Servers & Apigee API Proxy Bundles exported from Apigee OPDK/X/Hybrid.
+Validation is done by deploying a sample proxy which check if HOST & PORT is open from Apigee OPDK/X/Hybrid.
+
+> **NOTE**: Discovery of Targets in API Proxy & Sharedflows is limited to only parsing URL from `TargetEndpoint` & `ServiceCallout` Policy.
+
+> **NOTE**: Dynamic targets are **NOT** supported, Ex : `https://host.{request.formparam.region}.example.com}`
+
+## Disclaimer
+This is not an Officially Supported Google Product!
+
+## Pre-Requisites
+* python3.x
+* Please Install required Python Libs 
+
+```
+    python3 -m pip install requirements.txt
+```
+* Please fill in `input.properties`
+
+```
+[source]
+baseurl=http://34.131.144.184:8080/v1       # Apigee OPDK/Edge/X/Hybrid Base URL
+org=xxx-xxxx-xxx-xxxxx                      # Apigee OPDK/Edge/X/Hybrid Org
+auth_type=basic                             # API Auth type basic | oauth
+
+[target]
+baseurl=https://apigee.googleapis.com/v1    # Apigee OPDK/Edge/X/Hybrid Base URL
+org=xxx-xxxx-xxx-xxxxx                      # Apigee OPDK/Edge/X/Hybrid Org Id
+auth_type=oauth                             # API Auth type basic | oauth
+
+[csv]
+file=input.csv                              # Path to input CSV. Note: CSV needs HOST & PORT columns
+default_port=443                            # default port if port is not provided in CSV
+
+[validation]
+check_csv=true                              # 'true' to validate Targets in input csv
+check_proxies=true                          # 'true' to validate Proxy Targets else 'false'
```

Comment:
I don't know what this means

Reply:
You can also give targets which are not in apigee , but you may plan on having them in the future .

You do this by providing an input CSV file